docs/guides: add Dasharo Reviewers Guide (#384)
Dasharo/docs: docs/guides/dasharo-reviewers-guide.md

findings came from third-party benchmarks and community reports, which is
enough to know our current standing. Professional hardware reviewers should be
better at performing and analyzing measures than we are.

## Why We Ask To Fine Tune Settings

In general, we found out in our limited benchmarking that performance
out-of-the-box with Dasharo is measurably lower than performance with MSI
proprietary firmware, in the order of 6-8% on the benchmarked Core i5 12600K.
Lower performance was quickly tracked down to MSI using, by default, certain
processor operating parameters that are quite different from the values stated
on the Intel datasheets, resulting in alterations of the processor behavior
that allows it to sustain Turbo Boost with higher clock speeds and for longer
periods than otherwise possible, and thus appear to perform better overall.
After normalizing across both MSI proprietary firmware and Dasharo, the values
we found were different, and the benchmark difference was [reduced to around 2%
(21:19- 29:00)](https://vimeo.com/756437712).

We also found several review websites that have already written articles and

> analyses about this matter. Mainboard vendors use default values above those

Reviewer: ```suggestion
analysis about this matter. Mainboard vendors use default values above those
```

> analyses about this matter. Motherboard vendors use default values above those

Reviewer: ```suggestion
analyses about this matter. Mainboard vendors use default values above those
```

[GamersNexus about MSI default settings][gamersnexus2], [GamersNexus Z490
YouTube Review][gamersnexus3], [TechPowerUp][techpowerup], [Tom's
Hardware][tomshw]). Depending on whom you ask, using aggressive values by
default can be seen either as a convenience, since the end user gets more
performance without having to know how to configure the firmware, or as a form
of cheating, due to the end user usually having no idea about these changes and
that they are overclocking out-of-the-box. As for the time being, we're
sticking to Intel specification default values; these mainboard vendors'
custom optimizations hurts us because it makes the difference look much bigger
than it is.

What we want to showcase is how Dasharo (for this MSI port, coreboot +
EDKII UefiPayloadPkg) performance compares against the proprietary firmware
codebase. The only way to do so accurately is by ensuring that the hardware
operating values are the same in both Dasharo and the original firmware so that
the performance differences are not due to higher processor clock speeds thanks
to higher Power Limits, unlimited Turbo Boost times, or similar tricks. Note

> that we are okay with out-of-the-box comparisons, but that wouldn't be directly
> comparing the performance of both firmware codebases, which is our point. Thus,

Reviewer: ```suggestion
that we are okay with out-of-the-box comparisons,  but that wouldn't be an apples-to-apples
comparison of the performance of both firmware codebases, which is our point. Thus,
```

we found were different, and the benchmark difference was [reduced to around 2%
(21:19- 29:00)](https://vimeo.com/756437712).

We also found several review websites that have already written articles and
analyses about this matter. Motherboard vendors use default values above those
specified by Intel (technically overclocking) to produce better performance
results in benchmarks ([AnandTech][anandtech], [TechSpot][techspot],
[VideoCardz][videocardz], [Gamers Nexus about cheating][gamersnexus1],
[GamersNexus about MSI default settings][gamersnexus2], [GamersNexus Z490
YouTube Review][gamersnexus3], [TechPowerUp][techpowerup], [Tom's
Hardware][tomshw]). Depending on whom you ask, using aggressive values by
default can be seen either as a convenience, since the end user gets more
performance without having to know how to configure the firmware, or as a form
of cheating, due to the end user usually having no idea about these changes and
that they are overclocking out-of-the-box. As for the time being, we're
sticking to Intel specification default values; these motherboard vendors'
custom optimizations hurts us because it makes the difference look much bigger
than it is.

> Since what we want to showcase is how Dasharo (for this MSI port, coreboot +

Reviewer 2: ```suggestion
What we want to showcase is how Dasharo (for this MSI port, coreboot +
```

There is no subordinate clause here, no need for `since`

EDKII UefiPayloadPkg) performance compares against the proprietary firmware
codebase. The only way to do so accurately is by ensuring that the hardware
operating values are the same in both Dasharo and the original firmware so that
the performance differences are not due to higher processor clock speeds thanks
to higher Power Limits, unlimited Turbo Boost times, or similar tricks. Note
that we are okay with out-of-the-box comparisons, but that wouldn't be directly
comparing the performance of both firmware codebases, which is our point. Thus,
to make it fair, the hardware has to be tweaked to run with the same operating
values on both. As you can't change values on Dasharo unless you're willing to
recompile (or ask for a custom build with tweaked values), the easiest way is
by bringing MSI firmware values down to Dasharo/Intel levels.

## Find Your Processor Intel Default Parameters

> So far, 6 Processor configurable parameters significantly differ: PL1, PL2, PL1

Reviewer 2: ```suggestion
So far we've found six processor-configurable parameters that differ significantly: PL1, PL2, PL1
```

`So far` alone implies that more differences might be introduced.

and community reports, which is enough to have a good idea about where we're
currently standing. If anything, professional hardware reviewers should be
better at performing benchmarks and analysing them than us.

## Why We Ask To Fine Tune Settings

In general, what we found out in our limited benchmarking is that performance
out-of-the-box with Dasharo is measurabily lower than performance with MSI
propietary firmware, in the order of 6-8% on the benchmarked Core i5 12600K.
This was quickly tracked down to MSI using by default certain processor
operating parameters that are quite different from the values stated on the
Intel datasheets, resulting in alterations of the processor behavior that
allows it to sustain Turbo Boost with higher clock speeds and for longer time
periods than otherwise possible, and thus appear to perform better overall.
After normalizing across both MSI proprietary firmware and Dasharo the values
we found different, the benchmark difference was [reduced to around 2% (21:19-
29:00)](https://vimeo.com/756437712).

We also found out that several hardware review websites already wrote articles

> and analysis about this matter, as it is common that Motherboard vendors uses
> by default values above those specified by Intel (technically overclocking) to

Reviewer: ```suggestion
and analysis about this matter, as it is common that Motherboard vendors use
default values which are above those specified by Intel (technically overclocking) to
```

Reviewer: @pietrushnic neither this one.

Author: `which are` is reundant

cheating][gamersnexus1], [GamersNexus about MSI default
settings][gamersnexus2], [GamersNexus Z490 YouTube Review][gamersnexus3],
[TechPowerUp][techpowerup], [Tom's Hardware][tomshw]). Depending on who you ask,
using aggressive values by default can be seen either as a convenience, since
the end user gets more performance without having to know how to configure the
firmware, or as a form of cheating, due to the end user usually having no idea
about these changes and that they are overclocking out-of-the-box. As for the
time being we're sticking to Intel specification default values, these
motherboard vendors custom optimizations hurts us cause it makes the difference
look much bigger than it really is.

Since what we want to showcase is how Dasharo (for this MSI port, coreboot +
EDKII UefiPayloadPkg) performance compares against the propietary firmware codebase,
the only way to do so accurately is by making sure than the hardware operating
values are the same in both Dasharo and the original firmware so that the
performance differences are not due to higher processor clock speeds thanks to
higher Power Limits, unlimited Turbo Boost times, or similar tricks. Note than

> we don't mind about out-of-the-box comparisons, but that wouldn't be directly
> comparing the performance of both firmware codebases, which is our point  Thus,

Reviewer: ```suggestion
we don't mind about out-of-the-box comparisons, but that wouldn't apple-to-apple
comparison of the performance of both firmware codebases, which is our point. Thus,
```

Reviewer 2: ```suggestion
we don't mind out-of-the-box comparisons, but that wouldn't be an apples-to-apples
comparison of the performance of both firmware codebases, which is our point. Thus,
```

Reviewer: @pietrushnic I don't see this suggestion being applied

Author: @miczyg1 text was changed completely because of grammarly fixes. Those suggestions no longer apply.

Reviewer: Ok, opened a new suggestion...
